great_expectations/datasource/data_connector/runtime_data_connector.py

```diff
@@ -142,75 +120,61 @@ def get_batch_definition_list_from_batch_request(
+        (and data_references) that have been passed into the RuntimeDataConnector in this session, even though technically
+        only the most recent batch_data is available. This can be misleading. However, allowing the RuntimeDataConnector
+        to keep a record of all data_assets (and data_references) that have been passed in will allow for the proposed
+        behavior of RuntimeBatchRequest which will allow for paths and queries to be passed in as part of the BatchRequest.
+        Therefore this behavior will be revisited when the design of RuntimeBatchRequest and related classes are complete.
+        """
         self._validate_batch_request(batch_request=batch_request)
 
-        batch_identifiers: Optional[dict] = None
-        if batch_request.partition_request:
-            self._validate_batch_identifiers(
-                batch_identifiers=batch_request.partition_request.get(
-                    "batch_identifiers"
-                )
-            )
-            batch_identifiers = batch_request.partition_request.get("batch_identifiers")
-        if not batch_identifiers:
-            batch_identifiers = {}
+        batch_identifiers = batch_request.partition_request.get("batch_identifiers")
 
-        batch_definition_list: List[BatchDefinition]
+        self._validate_batch_identifiers(batch_identifiers=batch_identifiers)
 
+        batch_definition_list: List[BatchDefinition]
         batch_definition: BatchDefinition = BatchDefinition(
             datasource_name=self.datasource_name,
             data_connector_name=self.name,
-            data_asset_name=DEFAULT_DATA_ASSET_NAME,
+            data_asset_name=batch_request.data_asset_name,
             partition_definition=PartitionDefinition(batch_identifiers),
         )
-
-        if batch_definition_matches_batch_request(
-            batch_definition=batch_definition, batch_request=batch_request
-        ):
-            batch_definition_list = [batch_definition]
-        else:
-            batch_definition_list = []
-
+        batch_definition_list = [batch_definition]
+        self._update_data_references_cache(
+            batch_request.data_asset_name, batch_definition_list, batch_identifiers
+        )
         return batch_definition_list
 
-    def _map_data_reference_to_batch_definition_list(
-        self, data_reference: str, data_asset_name: Optional[str] = None
-    ) -> Optional[List[BatchDefinition]]:
-        if data_asset_name is None:
-            data_asset_name = DEFAULT_DATA_ASSET_NAME
-        return [
-            BatchDefinition(
-                datasource_name=self.datasource_name,
-                data_connector_name=self.name,
-                data_asset_name=data_asset_name,
-                partition_definition=PartitionDefinition(),
-            )
-        ]
-
-    def _map_batch_definition_to_data_reference(
+    def _update_data_references_cache(
```

Comment:
@Shinnnyshinshin : It looks like now we should be able too set the `data_asset_name` and then retrieve it as part of `get_available_data_assets()` on a near-future call.  Nice!!!

Reply:
🙇🏼